query-engine/metrics/src/lib.rs

```diff
@@ -34,165 +34,138 @@ use once_cell::sync::Lazy;
+static METRIC_RENAMES: Lazy<HashMap<&'static str, (&'static str, &'static str)>> = Lazy::new(|| {
     HashMap::from([
-        (MOBC_POOL_CONNECTIONS_OPENED_TOTAL, PRISMA_POOL_CONNECTIONS_OPENED_TOTAL),
-        (MOBC_POOL_CONNECTIONS_CLOSED_TOTAL, PRISMA_POOL_CONNECTIONS_CLOSED_TOTAL),
-        (MOBC_POOL_CONNECTIONS_OPEN, PRISMA_POOL_CONNECTIONS_OPEN),
-        (MOBC_POOL_CONNECTIONS_BUSY, PRISMA_POOL_CONNECTIONS_BUSY),
-        (MOBC_POOL_CONNECTIONS_IDLE, PRISMA_POOL_CONNECTIONS_IDLE),
-        (MOBC_POOL_WAIT_COUNT, PRISMA_CLIENT_QUERIES_WAIT),
-        (MOBC_POOL_WAIT_DURATION, PRISMA_CLIENT_QUERIES_WAIT_HISTOGRAM_MS),
+        (MOBC_POOL_CONNECTIONS_OPENED_TOTAL, ("prisma_pool_connections_opened_total", "The total number of pool connections opened")),
+        (MOBC_POOL_CONNECTIONS_CLOSED_TOTAL, ("prisma_pool_connections_closed_total", "The total number of pool connections closed")),
+        (MOBC_POOL_CONNECTIONS_OPEN, ("prisma_pool_connections_open", "The number of pool connections currently open")),
+        (MOBC_POOL_CONNECTIONS_BUSY, ("prisma_pool_connections_busy", "The number of pool connections currently executing datasource queries")),
+        (MOBC_POOL_CONNECTIONS_IDLE, ("prisma_pool_connections_idle", "The number of pool connections that are not busy running a query")),
+        (MOBC_POOL_WAIT_COUNT, ("prisma_client_queries_wait", "The number of datasource queries currently waiting for an free connection")),
+        (MOBC_POOL_WAIT_DURATION, ("prisma_client_queries_wait_histogram_ms", "The distribution of the time all datasource queries spent waiting for a free connection")),
     ])
 });
 
-// At the moment the histogram is only used for timings. So the bounds are hard coded here
-// The buckets are for ms
-pub(crate) const HISTOGRAM_BOUNDS: [f64; 10] = [0.0, 1.0, 5.0, 10.0, 50.0, 100.0, 500.0, 1000.0, 5000.0, 50000.0];
+pub fn setup() {
+    set_recorder();
+    initialize_metrics();
+}
 
-#[derive(PartialEq, Eq, Debug, Deserialize)]
-pub enum MetricFormat {
-    #[serde(alias = "json")]
-    Json,
-    #[serde(alias = "prometheus")]
-    Prometheus,
+static METRIC_RECORDER: Once = Once::new();
+
+fn set_recorder() {
+    METRIC_RECORDER.call_once(|| {
+        metrics::set_boxed_recorder(Box::new(MetricRecorder)).unwrap();
+    });
 }
 
-pub fn setup() {
-    set_recorder();
-    describe_metrics();
+/// Initialize metrics descriptions and values
+pub fn initialize_metrics() {
+    initialize_metrics_descriptions();
+    initialize_metrics_values();
 }
 
-// Describe all metric here so that every time for create
-// a new metric registry for a Query Instance the descriptions
-// will be in place
-pub fn describe_metrics() {
-    // counters
+/// Describe all first-party metrics that we record in prisma-engines. Metrics recorded by third-parties
+/// --like mobc-- are described by such third parties, but ignored, and replaced by the descriptions in the
+/// METRICS_RENAMES map.
+fn initialize_metrics_descriptions() {
```

Comment:
Nit: Maybe make more explicit that those are the "first party" metrics descriptions?